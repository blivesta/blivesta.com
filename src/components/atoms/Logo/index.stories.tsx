import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import Logo from '.';

const Wrapper = styled.div`
  width: 480px;
`;

storiesOf(`atoms|Logo`, module).add(`default`, () => (
  <Wrapper>
    <Logo />
  </Wrapper>
));
