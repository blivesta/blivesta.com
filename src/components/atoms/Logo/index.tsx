import * as React from 'react';
import styled from 'styled-components';

const PathLeft = styled.path`
  fill: #fff;
`;

const PathRight = styled.path`
  fill: ${({ theme }) => theme.colors.logo};
`;

const Logo = () => {
  return (
    <svg viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
      <PathLeft d="M0 0h512v512H0z" />
      <PathRight d="M280.448 0L152.545 127.712 280.48 255.424 152 383.712 280.448 512H512V0H280.448zm119.568 440L336 376l64.016-64L464 376l-63.984 64zm0-240L336 136l64.016-64L464 136l-63.984 64z" />
    </svg>
  );
};

export default Logo;
