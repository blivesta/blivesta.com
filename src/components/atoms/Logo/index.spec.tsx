import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Logo from '.';

describe('Logo', () => {
  it('snapshot', () => {
    const tree = shallow(<Logo />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
