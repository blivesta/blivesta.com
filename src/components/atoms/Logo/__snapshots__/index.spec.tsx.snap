// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo snapshot 1`] = `
<svg
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <styled.path
    d="M0 0h512v512H0z"
  />
  <styled.path
    d="M280.448 0L152.545 127.712 280.48 255.424 152 383.712 280.448 512H512V0H280.448zm119.568 440L336 376l64.016-64L464 376l-63.984 64zm0-240L336 136l64.016-64L464 136l-63.984 64z"
  />
</svg>
`;
